testing/test-nx-utils/src/lib/utils/nx.unit.test.ts

```diff
@@ -0,0 +1,84 @@
+import * as process from 'node:process';
+import { createTreeWithEmptyWorkspace } from 'nx/src/generators/testing-utils/create-tree-with-empty-workspace';
+import { describe, expect } from 'vitest';
+import { executorContext, registerPluginInWorkspace } from './nx';
+
+describe('executorContext', () => {
+  it('should create context for given project name', () => {
+    expect(executorContext('my-lib')).toStrictEqual({
+      cwd: process.cwd(),
+      isVerbose: false,
+      projectName: 'my-lib',
+      projectsConfigurations: {
+        projects: {
+          'my-lib': {
+            name: 'my-lib',
+            root: 'libs/my-lib',
+          },
+        },
+        version: 1,
+      },
+      root: '.',
+    });
+  });
+
+  it('should create context for given project options', () => {
+    expect(
+      executorContext({ projectName: 'other-lib', cwd: '<CWD>' }),
+    ).toStrictEqual({
+      cwd: '<CWD>',
+      isVerbose: false,
+      projectName: 'other-lib',
+      projectsConfigurations: {
+        projects: {
+          'other-lib': {
+            name: 'other-lib',
+            root: 'libs/other-lib',
+          },
+        },
+        version: 1,
+      },
+      root: '.',
+    });
+  });
+});
+
+describe('registerPluginInWorkspace', () => {
+  it('should register plugin name in workspace', () => {
+    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
+
+    registerPluginInWorkspace(tree, 'nx-plugin');
+
+    const nxJson = JSON.parse(tree.read('nx.json')?.toString() ?? '{}');
```

Comment:
🧪 Linear flow: There should be no need for conditional statements in tests. The test should know what is expected and not insert any default values such as `{}`. I would suggest using `!` here. Curious to hear what @matejchalk thinks.

This is applicable to multiple places.